src/utils.c

```diff
@@ -2152,39 +2153,56 @@ const gchar *utils_resource_dir(GeanyResourceDirType type)
 }
 
 
-void utils_start_new_geany_instance(const gchar *doc_path)
+// Attempts to get the path to the executable for this Geany instance
+static gchar *utils_find_self_executable(void)
 {
-	const gchar *command = is_osx_bundle() ? "open" : "geany";
-	gchar *exec_path = g_find_program_in_path(command);
+#if defined(MAC_INTEGRATION)
+	// GtkOSX integration library provides a wrapper to get path from bundle
+	if (is_osx_bundle())
+	{
+		gchar *exec_file = gtkosx_application_get_executable_path();
+		if (exec_file != NULL)
+			return exe_file;
+	}
+#endif
+
+	// Try using the installation directory
+	gchar *exe_file = g_build_filename(GEANY_PREFIX, "bin", "geany", NULL);
```

Comment:
You could declare the exe_file variable at the beginning of the function and use it both for the OS X and non-OS X case.